File: src/infrastructure/di/DummyBrokerageClient.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using core.Account;
using core.fs;
using core.fs.Adapters.Brokerage;
using core.fs.Adapters.Stocks;
using core.Shared;
using Microsoft.FSharp.Core;
using OptionChain = core.fs.Adapters.Options.OptionChain;

namespace di;

public class DummyBrokerageClient : IBrokerage
{
    public Task<FSharpResult<OAuthResponse, ServiceError>> ConnectCallback(string code)
    {
        throw new NotImplementedException();
    }

    public Task<string> GetOAuthUrl()
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<StockProfile,ServiceError>> GetStockProfile(UserState state, Ticker ticker)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<AccountTransaction[], ServiceError>> GetTransactions(UserState state, AccountTransactionType[] types)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<Unit, ServiceError>> OptionOrder(UserState state, string payload)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<BrokerageAccount,ServiceError>> GetAccount(UserState user)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<Unit,ServiceError>> CancelOrder(UserState user, string orderId)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<Unit,ServiceError>> BuyOrder(
        UserState user,
        Ticker ticker,
        decimal numberOfShares,
        decimal price,
        BrokerageOrderType type,
        BrokerageOrderDuration duration)
    {
        throw new NotImplementedException();
    }
    
    public Task<FSharpResult<Unit,ServiceError>> BuyToCoverOrder(
        UserState user,
        Ticker ticker,
        decimal numberOfShares,
        decimal price,
        BrokerageOrderType type,
        BrokerageOrderDuration duration)
    {
        throw new NotImplementedException();
    }
    
    public Task<FSharpResult<Unit,ServiceError>> SellShortOrder(
        UserState user,
        Ticker ticker,
        decimal numberOfShares,
        decimal price,
        BrokerageOrderType type,
        BrokerageOrderDuration duration)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<Unit,ServiceError>> SellOrder(
        UserState user,
        Ticker ticker,
        decimal numberOfShares,
        decimal price,
        BrokerageOrderType type,
        BrokerageOrderDuration duration)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<StockQuote,ServiceError>> GetQuote(UserState user, Ticker ticker)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<Dictionary<Ticker, StockQuote>, ServiceError>> GetQuotes(UserState user, IEnumerable<Ticker> tickers)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<SearchResult[], ServiceError>> Search(UserState state, SearchQueryType searchQueryType, string query, int limit)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<OptionChain, ServiceError>> GetOptionChain(UserState state, Ticker ticker)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<MarketHours, ServiceError>> GetMarketHours(UserState state, DateTimeOffset date)
    {
        throw new NotImplementedException();
    }

    public Task<FSharpResult<PriceBars, ServiceError>> GetPriceHistory(
        UserState state,
        Ticker ticker,
        PriceFrequency frequency,
        FSharpOption<DateTimeOffset> start,
        FSharpOption<DateTimeOffset> end)
    {
        throw new NotImplementedException();
    }

    public Task<OAuthResponse> RefreshAccessToken(UserState user) =>
        throw new NotImplementedException();

    public Task<OAuthResponse> GetAccessToken(UserState user)
    {
        throw new NotImplementedException();
    }
}

File: src/infrastructure/di.fs/DummyBrokerageClient.fs
namespace di

open System
open System.Collections.Generic
open System.Threading.Tasks
open core.Account
open core.fs
open core.fs.Adapters.Brokerage
open core.fs.Adapters.Stocks
open core.fs.Adapters.Options
open core.Shared

type DummyBrokerageClient() =
    
    interface IBrokerage with
        member _.ConnectCallback(code: string) : Task<Result<OAuthResponse, ServiceError>> =
            raise (NotImplementedException())
        
        member _.GetOAuthUrl() : Task<string> =
            raise (NotImplementedException())
        
        member _.GetStockProfile(state: UserState) (ticker: Ticker) : Task<Result<StockProfile, ServiceError>> =
            raise (NotImplementedException())
        
        member _.GetTransactions(state: UserState) (types: AccountTransactionType[]) : Task<Result<AccountTransaction[], ServiceError>> =
            raise (NotImplementedException())
        
        member _.OptionOrder(state: UserState) (payload: string) : Task<Result<unit, ServiceError>> =
            raise (NotImplementedException())
        
        member _.GetAccount(user: UserState) : Task<Result<BrokerageAccount, ServiceError>> =
            raise (NotImplementedException())
        
        member _.CancelOrder(state: UserState) (orderId: string) : Task<Result<unit, ServiceError>> =
            raise (NotImplementedException())
        
        member _.BuyOrder(state: UserState) (ticker: Ticker) (numberOfShares: decimal) (price: decimal) (orderType: BrokerageOrderType) (duration: BrokerageOrderDuration) : Task<Result<unit, ServiceError>> =
            raise (NotImplementedException())
        
        member _.BuyToCoverOrder(state: UserState) (ticker: Ticker) (numberOfShares: decimal) (price: decimal) (orderType: BrokerageOrderType) (duration: BrokerageOrderDuration) : Task<Result<unit, ServiceError>> =
            raise (NotImplementedException())
        
        member _.SellShortOrder(state: UserState) (ticker: Ticker) (numberOfShares: decimal) (price: decimal) (orderType: BrokerageOrderType) (duration: BrokerageOrderDuration) : Task<Result<unit, ServiceError>> =
            raise (NotImplementedException())
        
        member _.SellOrder(state: UserState) (ticker: Ticker) (numberOfShares: decimal) (price: decimal) (orderType: BrokerageOrderType) (duration: BrokerageOrderDuration) : Task<Result<unit, ServiceError>> =
            raise (NotImplementedException())
        
        member _.GetQuote(state: UserState) (ticker: Ticker) : Task<Result<StockQuote, ServiceError>> =
            raise (NotImplementedException())
        
        member _.GetQuotes(state: UserState) (tickers: seq<Ticker>) : Task<Result<Dictionary<Ticker, StockQuote>, ServiceError>> =
            raise (NotImplementedException())
        
        member _.Search(state: UserState) (searchQueryType: SearchQueryType) (query: string) (limit: int) : Task<Result<SearchResult[], ServiceError>> =
            raise (NotImplementedException())
        
        member _.GetOptionChain(state: UserState) (source: OptionChainSource) (ticker: Ticker) : Task<Result<OptionChain, ServiceError>> =
            raise (NotImplementedException())
        
        member _.GetMarketHours(state: UserState) (date: DateTimeOffset) : Task<Result<MarketHours, ServiceError>> =
            raise (NotImplementedException())
        
        member _.GetPriceHistory(state: UserState) (ticker: Ticker) (frequency: PriceFrequency) (start: DateTimeOffset option) (``end``: DateTimeOffset option) : Task<Result<PriceBars, ServiceError>> =
            raise (NotImplementedException())
        
        member _.RefreshAccessToken(user: UserState) : Task<OAuthResponse> =
            raise (NotImplementedException())
        
        member _.GetAccessToken(user: UserState) : Task<OAuthResponse> =
            raise (NotImplementedException())
